libs/logging/src/logger.ts

```diff
@@ -1,77 +1,30 @@
-import { LogEventId, getDetailsForEventId } from './log_event_ids';
-import { CLIENT_SIDE_LOG_SOURCES, LogSource } from './base_types/log_source';
-import {
-  LogDisposition,
-  LogDispositionStandardTypes,
-  LogLine,
-  LoggingUserRole,
-} from './types';
+import { LogEventId } from './log_event_ids';
+import { LogSource } from './base_types/log_source';
+import { LogLine, LoggingUserRole } from './types';
+import { LogData, BaseLogger } from './base_logger';
 
-export const LOGS_ROOT_LOCATION = '/var/log';
-export const LOG_NAME = 'vx-logs';
-export const FULL_LOG_PATH = `${LOGS_ROOT_LOCATION}/${LOG_NAME}.log`;
-
-const debug = makeDebug('logger');
-
-interface LogData extends Dictionary<string | boolean | number> {
-  message?: string;
-  disposition?: LogDisposition;
-}
-
-export class Logger {
+export class Logger extends BaseLogger {
   constructor(
-    private readonly source: LogSource,
-    private readonly kiosk?: KioskBrowser.Kiosk
-  ) {}
-
-  getSource(): LogSource {
-    return this.source;
+    source: LogSource,
+    private readonly getUserRole: () => Promise<LoggingUserRole>
+  ) {
+    super(source);
   }
 
-  async log(
+  async logAsCurrentUser(
     eventId: LogEventId,
-    user: LoggingUserRole,
     logData?: LogData,
     outerDebug?: (logLine: LogLine) => void
   ): Promise<void> {
-    const eventSpecificDetails = getDetailsForEventId(eventId);
-    const {
-      message = eventSpecificDetails.defaultMessage,
-      disposition = LogDispositionStandardTypes.NotApplicable,
-      ...additionalData
-    } = logData ?? {};
-    const logLine: LogLine = {
-      source: this.source,
-      eventId,
-      eventType: eventSpecificDetails.eventType,
-      user,
-      message,
-      disposition,
-      ...additionalData,
-    };
-    // If the caller is passing in a debug instance, and we are not in production log to the debugger rather then through the normal logging pipeline.
-    // This is to make logs more manageable in development, so a developer can toggle what logs to view with the normal debug namespaces.
-    /* istanbul ignore next - figure out how to test this */
-    if (outerDebug && process.env.NODE_ENV !== 'production') {
-      outerDebug(logLine);
-      return;
-    }
+    return this.log(eventId, await this.getUserRole(), logData, outerDebug);
+  }
 
-    if (CLIENT_SIDE_LOG_SOURCES.includes(this.source)) {
-      debug(logLine); // for internal debugging use log to the console
-      if (this.kiosk) {
-        await this.kiosk.log(
-          JSON.stringify({
-            timeLogInitiated: Date.now().toString(),
-            ...logLine,
-          })
-        );
-      }
-    } else {
-      // eslint-disable-next-line no-console
-      console.log(JSON.stringify(logLine));
-    }
+  static from(
+    baseLogger: BaseLogger,
+    getUserRole: () => Promise<LoggingUserRole>
+  ): Logger {
+    const logger = new Logger(baseLogger.getSource(), getUserRole);
+    logger.log = baseLogger.log;
```

Comment:
why is this line necessary?

Reply:
This is helpful in tests where we're passing in a mocked `BaseLogger` that is converted to a logger with `Logger.from`. I want the created `Logger` to still be mocked.